canmatrix/canmatrix.py

```diff
@@ -404,19 +408,15 @@ def phys2raw(self, value=None):
                 "Value {} is not valid for {}. Min={} and Max={}".format(
                     value, self, self.min, self.max)
                 )
-        raw_value = (value - self.offset) / self.factor
-
-        if not self.is_float:
-            raw_value = int(raw_value)
-        return raw_value
+        return (self.float_factory(value) - self.offset) / self.factor
 
-    def raw2phys(self, value, decodeToStr = False):
+    def raw2phys(self, value, decodeToStr=False):
         """Decode the given raw value (= as is on CAN)
         :param value: raw value
         :return: physical value (scaled)
         """
 
-        value = value * self.factor + self.offset
+        value = self.float_factory(value) * self.factor + self.offset
```

Comment:
The thing is that the caller may not know what is the `float_factory` class.

Reply:
How?  They are writing the program that is calling this.  If they want a `decimal.Decimal` to be used, they should pass it in instead of a `float`, for example.  If they pass a `float` then converting it to a `decimal.Decimal` isn't going to do a lot of good as the loss of accuracy has already occurred.

But I see you changed it so no hold up here.